src/shared/components/axisTicks/AxisTicksGenerator.tsx

```diff
@@ -0,0 +1,203 @@
+        return
+      }
+
+      case 'TickStep': {
+        onSetTickStep(arg as number)
+        return
+      }
+    }
+  }
+
+  return (
+    <Form.Element label={label} className="tick-generator-input">
+      <Grid.Row>
+        <Grid.Column widthXS={Columns.Twelve}>
+          <SelectGroup shape={ButtonShape.StretchToFit}>
+            <SelectGroup.Option
+              name="tick-generator"
+              id="radio_auto"
+              titleText="Auto"
+              active={!showInputs}
+              onClick={handleChooseAuto}
+              value="Auto"
+            >
+              Auto
+            </SelectGroup.Option>
+            <SelectGroup.Option
+              name="tick-generator"
+              id="radio_custom"
+              titleText="Custom"
+              active={showInputs}
+              onClick={handleChooseCustom}
+              value="Custom"
+            >
+              Custom
+            </SelectGroup.Option>
+          </SelectGroup>
+        </Grid.Column>
+      </Grid.Row>
+      {showInputs &&
+        (isTimeColumn ? (
+          <TimeTicksOptions
+            axisName={axisName}
+            tickOptions={generateAxisTicks}
+            totalTicks={totalTicks}
+            tickStart={tickStart}
+            tickStep={tickStep}
+            setOptions={onSetTicksGeneratorOptions}
+          />
+        ) : (
+          <ValueTicksOptions
+            axisName={axisName}
+            tickOptions={generateAxisTicks}
+            totalTicks={totalTicks}
+            tickStart={tickStart}
+            tickStep={tickStep}
+            setOptions={onSetTicksGeneratorOptions}
+          />
+        ))}
+    </Form.Element>
+  )
+}
+
+const mstp = (state: AppState, ownProps: AxisTicksGeneratorProps) => {
+  const {axisName} = ownProps
+  const timeMachine = getActiveTimeMachine(state)
+  const properties = get(timeMachine, 'view.properties', {})
+
+  const generateAxisTicks =
+    properties[`generate${capitalize(axisName)}AxisTicks`]
+  const totalTicks = properties[`${axisName}TotalTicks`]
+  const tickStart = properties[`${axisName}TickStart`]
+  const tickStep = properties[`${axisName}TickStep`]
+
+  let hasTotalTicks =
+    typeof totalTicks === 'number' && totalTicks === totalTicks
+  let hasTickStart = typeof tickStart === 'number' && tickStart === tickStart
+  let hasTickStep = typeof tickStep === 'number' && tickStep === tickStep
+
+  if (Array.isArray(generateAxisTicks)) {
+    generateAxisTicks.forEach(tickOption => {
+      if (tickOption === `${axisName.toLowerCase()}TotalTicks`) {
+        hasTotalTicks = true
+      }
+      if (tickOption === `${axisName.toLowerCase()}TickStart`) {
+        hasTickStart = true
+      }
+      if (tickOption === `${axisName.toLowerCase()}TickStep`) {
+        hasTickStep = true
+      }
+    })
+  }
+
+  return {
+    generateAxisTicks: Array.isArray(generateAxisTicks)
+      ? generateAxisTicks
+      : [],
+    totalTicks: hasTotalTicks ? totalTicks : '',
```

Comment:
This will be feed into a user input field. An empty string represents "no user input".

Even if we don't want to use an empty string here, there is a subtle distinction between "lack of value" versus 0. "lack of value" means, to Giraffe, please calculate the ticks as you see fit. 0 means exactly that -- 0. For example, 0 for total ticks would mean don't label the axis with any ticks. Or, for tick start, 0 would mean start at the value 0 on the axis. (Tick step can also be 0 -- Giraffe handles that gracefully.)

And, so, we use an empty string here to denote "lack of value" which is not the same as 0.